engine/src/main/java/com/datatorrent/stram/plan/logical/LogicalPlan.java

```diff
@@ -1359,7 +1409,8 @@ private void addDAGToCurrentDAG(ModuleMeta moduleMeta)
     String name;
     for (OperatorMeta operatorMeta : subDag.getAllOperators()) {
       name = subDAGName + MODULE_NAMESPACE_SEPARATOR + operatorMeta.getName();
-      this.addOperator(name, operatorMeta.getOperator());
+      Operator op = this.addOperator(name, operatorMeta.getOperator());
```

Comment:
Are we keeping module DAG after adding it's operators to the final DAG? If not, there is no meta object sharing between DAGs.
Introducing copy constructor will allow to share some of final collection references. As operator reference is shared between meta objects, should other references be also shared?